File: Support/Templates/vb/Files/DynamicRootBindingList.vb
Imports Csla
Imports Csla.Security

<Serializable()>
Public Class DynamicRootBindingList
  Inherits DynamicBindingListBase(Of DynamicRoot)

  Protected Overrides Function AddNewCore() As Object

    Dim item As DynamicRoot = DataPortal.Create(Of DynamicRoot)()
    Add(item)
    Return item

  End Function

  Private Shared Sub AddObjectAuthorizationRules()
    'TODO: add authorization rules
    'AuthorizationRules.AllowGet(GetType(DynamicRootBindingList), "Role")
    'AuthorizationRules.AllowEdit(GetType(DynamicRootBindingList), "Role")
  End Sub

  <Fetch>
  Private Sub Fetch()

    ' TODO: load values
    RaiseListChangedEvents = False
    Dim listData As Object = Nothing
    For Each oneItemData As Object In CType(listData, List(Of Object))
      Add(DataPortal.Fetch(Of DynamicRoot)(oneItemData))
    Next
    RaiseListChangedEvents = True

  End Sub

End Class

File: Support/Templates/cs/Files/DynamicRootBindingList.cs
using System;
using System.Collections.Generic;
using Csla;

namespace Templates
{
  [Serializable]
  public class DynamicRootBindingList :
    DynamicBindingListBase<DynamicRoot>
  {
    protected override object AddNewCore()
    {
      DynamicRoot item = DataPortal.Create<DynamicRoot>();
      Add(item);
      return item;
    }

    private static void AddObjectAuthorizationRules()
    {
      // TODO: add authorization rules
      //AuthorizationRules.AllowGet(typeof(DynamicRootBindingList), "Role");
      //AuthorizationRules.AllowEdit(typeof(DynamicRootBindingList), "Role");
    }

    [Fetch]
    private void Fetch()
    {
      // TODO: load values
      RaiseListChangedEvents = false;
      object listData = null;
      foreach (var item in (List<object>)listData)
        Add(DataPortal.Fetch<DynamicRoot>(item));
      RaiseListChangedEvents = true;
    }
  }
}